be/src/util/mem_info.cpp

```diff
@@ -80,10 +82,14 @@ void MemInfo::init() {
     }
 
     LOG(INFO) << "Physical Memory: " << PrettyPrinter::print(_s_physical_mem, TUnit::BYTES);
-
     _s_initialized = true;
 }
 
+int64_t MemInfo::mem_limit() {
+    bool is_percent = true;
+    return ParseUtil::parse_mem_spec(config::mem_limit, -1, MemInfo::physical_mem(), &is_percent);
```

Comment:
The ParseUtil::parse_mem_spec() should not be called every time we call `mem_limit()`